inc/file.class.php

```diff
@@ -68,70 +68,31 @@ public static function getMenuPicture() {
       return 'fa-file';
    }
 
-   /**
-    * @see CommonDBTM::addNeededInfoToInput()
-    */
    public function addNeededInfoToInput($input) {
       $input['entities_id'] = $_SESSION['glpiactive_entity'];
 
       return $input;
    }
 
-   /**
-    * (non-PHPdoc)
-    * @see CommonDBTM::prepareInputForAdd()
-    */
    public function prepareInputForAdd($input) {
-      // Find the added file
-      if (isset($_POST['_file'][0]) && is_string($_POST['_file'][0])) {
-         // from GLPI UI
-         $actualFilename = $_POST['_file'][0];
-         $uploadedFile = GLPI_TMP_DIR . "/" . $_POST['_file'][0];
-      } else {
-         // from API
-         if (!isset($_FILES['file'])) {
-            Session::addMessageAfterRedirect(__('No file uploaded', "flyvemdm"));
-            return false;
-         }
-
-         if (!$_FILES['file']['error'] == 0) {
-            if (!$_FILES['file']['error'] == 4) {
-               Session::addMessageAfterRedirect(__('File upload failed', "flyvemdm"));
-            }
-            return false;
-         }
-
-         $destination = GLPI_TMP_DIR . '/' . $_FILES['file']['name'];
-         if (is_readable($_FILES['file']['tmp_name']) && !is_readable($destination)) {
-            // Move the file to GLPI_TMP_DIR
-            if (!is_dir(GLPI_TMP_DIR)) {
-               Session::addMessageAfterRedirect(__("Temp directory doesn't exist"), false, ERROR);
-               return false;
-            }
-
-            // With GLPI < 9.2, the file was not moved by the API
-            if (!move_uploaded_file($_FILES['file']['tmp_name'], $destination)) {
-               return false;
-            }
-         }
-
-         $actualFilename = $_FILES['file']['name'];
-         $uploadedFile = $destination;
+      list($actualFilename, $uploadedFile) = $this->getUploadedFile(true);
+      if ($actualFilename === null) {
+         return false;
       }
 
       if (!isset($input['entities_id'])) {
          $input['entities_id'] = $_SESSION['glpiactive_entity'];
       }
 
       if (!isset($actualFilename)) {
-         Session::addMessageAfterRedirect(__('File uploaded without name', "flyvemdm"));
+         Session::addMessageAfterRedirect(__('File uploaded without name', 'flyvemdm'));
          return false;
       }
       $input['source'] = $input['entities_id'] . "/" . uniqid() . "_" . basename($uploadedFile);
       $destination = FLYVEMDM_FILE_PATH . "/" . $input['source'];
       $this->createEntityDirectory(dirname($destination));
       if (!rename($uploadedFile, $destination)) {
-         if (!is_writable(dirname($destination))) {
+         if (!file_exists(dirname($destination))) {
```

Comment:
I don't like to use file_exists function for check if a file is uploaded because that function returns true even if the file doesn't exist. It's behavior is to check against files and dirs and if the source input is empty it will return true, so it's better to use is_file or is_readable IMO